include/sys/dmu.h

```diff
@@ -754,7 +755,8 @@ typedef struct dmu_object_info {
 	uint8_t doi_checksum;
 	uint8_t doi_compress;
 	uint8_t doi_nblkptr;
-	uint8_t doi_pad[4];
+	uint8_t doi_count;
```

Comment:
count of what?  see comment about dn_count.  consider renaming, or at least add a comment here.